test/unit-test/mqtt_agent_utest.c

```diff
@@ -1104,22 +1147,23 @@ void test_MQTTAgent_CommandLoop_with_empty_command_queue( void )
 {
     MQTTStatus_t mqttStatus;
     MQTTAgentContext_t mqttAgentContext;
-    MQTTAgentCommandFuncReturns_t pReturnFlags;
 
     setupAgentContext( &mqttAgentContext );
 
-    mqttAgentContext.mqttContext.connectStatus = MQTTConnected;
+    mqttAgentContext.mqttContext.connectStatus = MQTTNotConnected;
     pReturnFlags.addAcknowledgment = false;
     pReturnFlags.runProcessLoop = true;
-    pReturnFlags.endLoop = true;
+    pReturnFlags.endLoop = false;
 
     MQTTAgentCommand_ProcessLoop_ExpectAnyArgsAndReturn( MQTTSuccess );
     MQTTAgentCommand_ProcessLoop_ReturnThruPtr_pReturnFlags( &pReturnFlags );
-    MQTT_ProcessLoop_IgnoreAndReturn( MQTTSuccess );
+    MQTT_ProcessLoop_Stub( MQTT_ProcessLoop_stub );
+
+    MQTTAgentCommand_ProcessLoop_ExpectAnyArgsAndReturn( MQTTBadParameter );
```

Comment:
It really shouldn't be returning bad parameter when all the parameters are valid. Receive failed makes more sense